brainiak/reprsimil/brsa.py

```diff
@@ -87,106 +176,205 @@ class BRSA(BaseEstimator):
+    space_smooth_range: float. Default: None
+        The distance (in unit the same as what
         you would use when supplying the spatial coordiates of
         each voxel, typically millimeter) which you believe is
         the maximum range of the length scale parameter of
         Gaussian Process defined over voxel location. This is
         used to impose a half-Cauchy prior on the length scale.
         If not provided, the program will set it to half of the
         maximum distance between all voxels.
-    inten_smooth_range: the difference in image intensity which
+    inten_smooth_range: float. Default: None
+        The difference in image intensity which
         you believe is the maximum range of plausible length
         scale for the Gaussian Process defined over image
         intensity. Length scales larger than this are allowed,
         but will be penalized. If not supplied, this parameter
         will be set to half of the maximal intensity difference.
-    tau_range: the reasonable range of the standard deviation
-        of the Gaussian Process. Since the Gaussian Process is
-        imposed on the log(SNR), this range should not be too
-        large. 10 is a pretty loose range. This parameter is
-        used in a half-Cauchy prior on the standard deviation
-    init_iter: how many initial iterations to fit the model
+    tau_range: float. Default: 5.0
+        The reasonable range of the standard deviation
+        of log(SNR). This range should not be too
+        large. 5 is a loose range.
+        When a Gaussian Process is imposed on the log(SNR),
+        this parameter is used in a half-Cauchy prior
+        on the standard deviation, or an inverse-Gamma prior
+        on the variance of the GP.
+    tau2_prior: callable[[float, int, float], [float, float]],
+        Default: prior_GP_var_inv_gamma.
+        Can be prior_GP_var_inv_gamma or prior_GP_var_half_cauchy,
+        or a custom function.
+        The function which impose a prior for tau^2, the variance of the
+        GP prior on log(SNR), and returns the MAP estimate of tau^2.
+        It can be either prior_GP_var_inv_gamma for inverse-Gamma
+        or prior_GP_var_half_cauchy for half-Cauchy.
+        But half-Cauchy prior is actually imposed on tau.
+        tau_range still describes the range of tau in the prior
+        in both cases. Inverse-Gamma prior
+        penalizes for very small tau and very large tau,
+        while half-Cauchy prior only penalizes for very large tau.
+        Both the functions are part of brsa module, which you need
+        to import together with BRSA in order to use.
+        For example::
+        from brainiak.reprsimil.brsa import BRSA, prior_GP_var_inv_gamma
+        brsa = BRSA(tau2_prior=prior_GP_var_inv_gamma)
+        Or, if you do are happy with the default inverse-Gamma prior,
+        then you can ignore this argument::
+        from brainiak.reprsimil.brsa import BRSA
+        brsa = BRSA()
```

Comment:
I suggest you keep only this usage example and remove the examples from the two functions.

Furthermore, you should remove the descriptions of the functions, which are already in the function docstrings, where they belong. Make it easier for users to jump to the functions in the HTML docs by marking the functions (e.g., `` `.prior_GP_var_half_cauchy` ``).